### PR TITLE
Adjust sasconf based on BIGREGION

### PR DESCRIPTION
**Pull request by RajalakshmiS**
__BIGREGION__ is adjusted based on input to use
appropriate mapping region for 32bit platforms.

2016-08-17  Rajalakshmi Srinivasaraghavan  <raji@linux.vnet.ibm.com>

	* src/sasconf.h: Check if __BIGREGION__ is zero and undefine it.

Signed-off-by:  Rajalakshmi Srinivasaraghavan  <raji@linux.vnet.ibm.com>

### CODE REVIEW

File: src/sasconf.h [hunk 1]
```diff
@@ -14,7 +14,11 @@
 
 #include <sys/param.h>
 
-#define __BIGREGION__ 1
+#if defined(__BIGREGION__) && __BIGREGION__ == 0
+# undef __BIGREGION__
+#else
+# define __BIGREGION__ 1
+#endif
```
Comment (tuliom) [inline src/sasconf.h]: If you make this change __BIGREGION__ will always be disabled unless someone manually specifies that at build time.
Isn't it better to enable or disable it automatically depending on the host (i.e. configure --host) value?

Comment (RajalakshmiS) [inline src/sasconf.h]: @tuliom, With this change, by default we will be using the values in #ifdef __BIGREGION__ 
For smaller/older systems , we have to use CFLAGS="-D__BIGREGION__=0" CXXFLAGS="-D__BIGREGION__=0" during configure.

Comment (tuliom) [inline src/sasconf.h]: Ooops.  You're right.
But my second point is still valid: shouldn't we set this macro automatically by default?
The person building SPHDE shouldn't need to know about SPHDE internals in order to build it...

File: configure.ac [hunk 2]
```diff
@@ -69,6 +69,13 @@ x86_64*)
   ;;
   esac
 ;;
+*64*)
+;;
+*)
+AC_MSG_WARN([
+*** __BIGREGION__ is not defined.
+*** Adjust the address limits in sasconf.h as required.])
+;;
```
Comment (tuliom) [inline configure.ac]: I still believe we should define __BIGREGION__ automatically based on the information provided by AC_CANONICAL_BUILD.  It shouldn't be complex and we won't be exposing SPHDE internals to the users.

Comment (RajalakshmiS) [inline configure.ac]: @tuliom Yes it is disabled by default for 32 bit. Just added a warning to inform that if the address limits differ, sasconf change would be required.

Comment (tuliom) [inline configure.ac]: I'm sorry.  I missed that.
LGTM.

**Pull request closed without merging**